model_compression_toolkit/target_platform_capabilities/schema/schema_compatability.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2025 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import Any
+
+import model_compression_toolkit.target_platform_capabilities.schema.v1 as schema_v1
+import model_compression_toolkit.target_platform_capabilities.schema.v2 as schema_v2
+import model_compression_toolkit.target_platform_capabilities.schema.mct_current_schema as schema
+
+ALL_SCHEMA_VERSIONS = [schema_v1]
+
+
+def is_tpc_instance(tpc_obj_or_path: Any) -> bool:
+    """
+    Checks if the given object is an instance of a TargetPlatformCapabilities
+    :param tpc_obj_or_path: Object to check its type
+    :return: True if the given object is an instance of a TargetPlatformCapabilities, False otherwise
+    """
+    all_tpc_types = [s.TargetPlatformCapabilities for s in ALL_SCHEMA_VERSIONS]
+    return type(tpc_obj_or_path) in all_tpc_types
+
+
+def _schema_v1_to_v2(tpc: schema_v1.TargetPlatformCapabilities) -> schema.TargetPlatformCapabilities:
+    """
+    Converts given tpc of schema version 1 to schema version 2
+    :return: TargetPlatformCapabilities instance of of schema version 2
+    """
+    raise NotImplementedError("Once schema v2 is implemented, add necessary adaptations to _schema_v1_to_v2 function")
+    return schema.TargetPlatformCapabilities(default_qco=tpc.default_qco,
+                                             operator_set=tpc.operator_set,
+                                             fusing_patterns=tpc.fusing_patterns,
+                                             tpc_minor_version=tpc.tpc_minor_version,
+                                             tpc_patch_version=tpc.tpc_patch_version,
+                                             tpc_platform_type=tpc.tpc_platform_type,
+                                             add_metadata=tpc.add_metadata)
+
+
+def tpc_to_current_schema_version(tpc: schema.TargetPlatformCapabilities):
```

Comment:
the type is not necessarily of `schema.TargetPlatformCapabilities` right? can be any schema version `TargetPlatformCapabilities`